Client/Helpers/FxALoginHelper.swift

```diff
@@ -155,6 +155,12 @@ class FxALoginHelper {
             account.updateProfile()
         }
         
+        if AppConstants.MOZ_ENABLE_LEANPLUM && AppConstants.MOZ_FXA_LEANPLUM_AB_PUSH_TEST {
+            // If Leanplum A/B push notification tests are enabled, defer to them for
+            // displaying the pre-push dialog.
+            return readyForSyncing()
```

Comment:
Looking through the code calling `readyForSyncing` seemed like the correct thing to do. From my tests, the account transitions back to settings and starts syncing. Leanplum then shows the pre-push dialog where the user can choose to enable or disabled push notifications.

@jhugman Do you see anything inherently wrong with doing this?

Reply:
If you're going to not calling `requestUserNotifications` here, then when are we registering with Autopush (to make a push registration)?